### PR TITLE
fix Container.clipBehaviour, clips the shadow in decoration

### PR DESCRIPTION
**Pull request by sahandevs**
## Description

fix Container.clipBehaviour, clips the shadow in BoxDecoration if provided.
fixed by changing the clip order

## Tests

I added the following tests:

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by goderbauer** [comment]
This PR is missing a test. Can you please add one to show what this is fixing and to ensure that we never break it again? Thanks.

**Comment by sahandevs** [comment]
@goderbauer I added a golden test. here is the snapshot:

widgets container clipBehaviour with shadow

@Piinks I don't have access to the Flutter Gold dashboard. not sure how to submit this snapshot.

**Comment by Piinks** [comment]
Hey @sahandevs I can help you check in the images. Can you update this to fix the analyzer?

**Comment by skia-gold** [comment]
Gold has detected about 1 untriaged digest(s) on patchset 5.
View them at https://flutter-gold.skia.org/cl/github/64362

**Comment by skia-gold** [comment]
Gold has detected about 1 untriaged digest(s) on patchset 6.
View them at https://flutter-gold.skia.org/cl/github/64362

File: packages/flutter/test/widgets/container_test.dart [hunk 1]
```diff
@@ -555,6 +555,37 @@ void main() {
...
+      RepaintBoundary(
+        child: Padding(
+          padding: const EdgeInsets.all(30.0),
+          child: container,
+      )),
+    );
+
+    await expectLater(
+      find.byType(RepaintBoundary),
+      matchesGoldenFile('widgets.container.clipBehaviour.with.shadow.png'),
```
Comment (Piinks) [inline packages/flutter/test/widgets/container_test.dart:586]:
```suggestion
      matchesGoldenFile('container.clipBehaviour.with.shadow.png'),
```
The library name does not need to be included. It is handled manually. :)

**Comment by skia-gold** [comment]
Gold has detected about 1 untriaged digest(s) on patchset 7.
View them at https://flutter-gold.skia.org/cl/github/64362

**Review by Piinks — APPROVED** [review]
LGTM! Thank you for the contribution!

**Pull request merged**